packages/ckeditor5-core/src/multicommand.js

```diff
@@ -56,23 +58,44 @@ export default class MultiCommand extends Command {
 	}
 
 	/**
-	 * Executes the first of it registered child commands.
+	 * Executes the first enabled command which has the highest priority of all registered child commands.
 	 *
 	 * @returns {*} The value returned by the {@link module:core/command~Command#execute `command.execute()`}.
 	 */
 	execute( ...args ) {
 		const command = this._getFirstEnabledCommand();
 
-		return command != null && command.execute( args );
+		return !!command && command.execute( args );
+	}
+
+	/**
+	 * Inserts command definition at correct index by priority so registered commands are always sorted from lowest priority to highest
+	 *
+	 * @param {Object} newCommandDefinition Object with `command` and `priority` properties
+	 * Object
+	 * @returns {undefined}
+	 * @private
+	 */
+	_insertCommandDefinitionByPriority( newCommandDefinition ) {
+		for ( let i = 0; i <= this._childCommandsDefinitons.length; i++ ) {
+			const registeredCommand = this._childCommandsDefinitons[ i ];
+
+			if ( !registeredCommand || priorities.get( registeredCommand.priority ) >= priorities.get( newCommandDefinition.priority ) ) {
+				this._childCommandsDefinitons.splice( i, 0, newCommandDefinition );
+
+				break;
+			}
+		}
 	}
 
 	/**
 	 * Registers a child command.
 	 *
 	 * @param {module:core/command~Command} command
+	 * @param {String|Number} priority Priority of command. Command with highest priority will be executed over others.
 	 */
-	registerChildCommand( command ) {
-		this._childCommands.push( command );
+	registerChildCommand( command, priority = 'normal' ) {
```

Comment:
To keep it consistent with the Emitter interface I would use an `option` object.